### PR TITLE
RES-1897 Improve query efficiency to remove Sentry errors.

### PR DESCRIPTION
**Pull request by edwh**
Fix N+1 queries:

1. https://the-restart-project.sentry.io/issues/3980915929
2. https://the-restart-project.sentry.io/issues/3808301382

We fix 2 by caching the category.  
* So far as I can tell there isn't a nice way to cache relationships. 
* If you put a cache call inside the getter then it moans about not being able to cache PDO object.  
* So we have a bit of code to cache before called the getter.  
* This is a bit hacky but reduces the number of DB calls for `/outbound/info/group/496/consume` from 766 to 61 - so it's a significant perf gain.

Also add some tests to improve coverage.  This fixes a bug in the scope for the groups visible to network coordinators.

### CODE REVIEW

**Review by ngm — COMMENTED** [review]
Looks good, however do we have tests around the calendar export and the stats changes?  Want to be cautious around changes to those.

Could you let Monique know about those stats caching changes too please, so she's aware of them next time needs to work on that code.

**Comment by edwh** [comment]
> Looks good, however do we have tests around the calendar export and the stats changes? Want to be cautious around changes to those.

We do - `CalendarTest`

> 
> Could you let Monique know about those stats caching changes too please, so she's aware of them next time needs to work on that code.

Will do.

File: app/Group.php [hunk 1]
```diff
@@ -431,12 +392,6 @@ public function getNextUpcomingEvent()
         return $event->first();
     }
 
-    public function scopeRequiresModeration($query)
```
[the remark this answers is not in the document]
Comment (edwh) [reply app/Group.php:434]: I looked fairly hard (search for text not just search by reference).  I've checked again, and I can't see anything.

File: app/Http/Controllers/API/GroupController.php [hunk 2]
```diff
@@ -469,7 +469,7 @@ private function getUser() {
         $user = Auth::user();
 
         if (!$user) {
-            $user = auth('api')->user();
+        $user = auth('api')->user();
```
[the remark this answers is not in the document]
Comment (edwh) [reply app/Http/Controllers/API/GroupController.php]: True.  Fixed.

**Review by ngm — APPROVED** [review]

**Pull request merged**